app/services/check_answers/relevant_section_finder_service.rb

```diff
@@ -22,9 +22,7 @@ def call
         end
       end
 
-      dictionary = nested_pairs.flatten.map { [_1[:key], _1[:value]] }.to_h
-
-      dictionary.fetch(@step.to_s)
+      nested_pairs.flatten.detect { _1[:key] == @step }&.fetch(:value)
```

Comment:
Some sort of small regression has been added here: Previously if you went: CheckAnswersScreen->VehicleScreen->VehicleDetailsScreen->CheckAnswerScreen, the final redirect would add an `#assets-section` anchor to the URL. Now when I try it the anchor is `#-section` and therefore broken.

Note that  CheckAnswersScreen->VehicleScreen->CheckAnswerScreen _does_ still end up with the correct anchor, so it's something to do with multi-step check answer loops.

Reply:
good spot, but sadly wasn't caused by this specific change. Rework to come